### PR TITLE
Adding options and spelling check

### PR DESCRIPTION
**Pull request by benoitm974**
Adding options
 - add a copy button
 - always copy result in clipboard
 - choose the openai model

add a new prompt and corresponding context menu for spelling and grammar check

### CODE REVIEW

**Comment by benoitm974** [comment]
also:
 - user innerText rather than innerHTML to avoid code interpreting with GPT-4 result
 - reduce the requested permission
 - edited the Readme to reflect changes
 - change a a genAI logo to avoit issue with chatGPT logo
 The goal would be to push this version to the chrome extension store (currently in review)

File: README.md [hunk 1]
```diff
@@ -19,9 +23,11 @@ To use this extension:
 
 - Obtain a ChatGPT API key by following the instructions on the [OpenAI website](https://platform.openai.com/docs/api-reference/authentication).
 - Click on the ChatGPT extension icon in the Chrome toolbar.
+For code improvement o spelling and grammar:
```
Comment (acumino) [inline README.md]:
```suggestion
For code improvement or checking spelling and grammar:
```

File: README.md [hunk 2]
```diff
@@ -19,9 +23,11 @@ To use this extension:
...
 - Select the code that you want to improve
 - Enter your API key in the provided field and click "Save".
-- To generate suggestions, select text on a webpage and right-click to bring up the context menu. Click "Suggest Improvement" and wait for the suggestion to appear.
+- To generate suggestions, select text on a webpage and right-click to bring up the context menu. Click "Suggest Improvement" or "Spelling and grammar" and wait for the suggestion to appear.
```
Comment (acumino) [inline README.md]:
```suggestion
- To generate suggestions, select text on a webpage and right-click to bring up the context menu. Click "Suggest Improvement" or "Spelling and Grammar" and wait for the suggestion to appear.
```

File: background.js [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+
+    const api_URL = "https://api.openai.com/v1";
+    chrome.storage.sync.set({"chatGPT_api_URL": api_URL}, function() {
+        console.log("API url saved - using: " + api_URL);
+    });
+
     //create context menu
     chrome.contextMenus.create({
         id: "wording",
+        title: "Spelling & Grammar",
```
Comment (acumino) [inline background.js]:
```suggestion
        title: "Spelling and Grammar",
```

File: manifest.json [hunk 4]
```diff
@@ -1,15 +1,18 @@
 {
   "name": "ChatGPT-Chrome-Extension",
-  "description": "This extension helps in code review by providing code suggestions through ChatGPT API",
-  "version": "0.0.1",
+  "description": "This extension helps in code review, spelling and grammar through ChatGPT API",
+  "version": "0.0.3",
```
Comment (acumino) [inline manifest.json]:
```suggestion
  "version": "0.0.2",
```

File: manifest.json [hunk 5]
```diff
@@ -1,15 +1,18 @@
 {
   "name": "ChatGPT-Chrome-Extension",
-  "description": "This extension helps in code review by providing code suggestions through ChatGPT API",
-  "version": "0.0.1",
+  "description": "This extension helps in code review, spelling and grammar through ChatGPT API",
```
Comment (acumino) [inline manifest.json]:
```suggestion
  "description": "This extension helps in code review and checking spelling and grammar through ChatGPT API",
```

File: background.js [hunk 6]
```diff
@@ -1,22 +1,34 @@
...
-        if (chrome.runtime.lastError) {
-            console.error(chrome.runtime.lastError);
-            return;
-        }
-        const api_key = result.chatGPT_api_key;
 
-        chrome.tabs.sendMessage(tab.id, { api_key: api_key, info: info });
+    chrome.storage.sync.get({
+        selectedModel: false,
+        chatGPT_api_key: false,
```
Comment (acumino) [inline background.js:28]: If `chatGPT_api_key` is not set an error should be thrown. How it was done in the old code.

File: content.js [hunk 7]
```diff
@@ -40,6 +61,14 @@ chrome.runtime.onMessage.addListener(function (message, sender, sendResponse) {
         });
 });
 
+function copyTextToClipboard(text) {
+    navigator.clipboard.writeText(text).then(function() {
+      //console.log('Copying to clipboard was successful!');
```
Comment (acumino) [inline content.js:66]: Not needed.
```suggestion
      //console.log('Copying to clipboard was successful!');
```

File: popup.html [hunk 8]
```diff
@@ -7,14 +7,45 @@
...
+    </div>
+    <h2>Extension Options</h2>
+
+    <div>
+        <label class="list">
+            <form>
+                <label for="model-select">Choose a model</label>
+                <select id="model-select" name="model-select">
+                </select>
+             </form>
```
Comment (acumino) [inline popup.html:26]: There should be a possible list of models to select.

**Review by acumino — CHANGES_REQUESTED** [review]

File: popup.html (hunk 8 above)
Comment (benoitm974) [reply popup.html:26]: Not it is populated once you put you api_key directly from available model from api

File: background.js (hunk 6 above)
Comment (acumino) [reply background.js:28]: What about this?

**Comment by acumino** [comment]
Thank you! @benoitm974

**Pull request merged**